src/lib/TurboConfirm.js

```diff
@@ -0,0 +1,104 @@
+import ConfirmationController from './ConfirmationController'
+
+export class TurboConfirm {
+  #controller
+  #config = {
+    dialogSelector: '#confirm',
+    activeClass: 'modal--active',
+    acceptSelector: '#confirm-accept',
+    denySelector: '.confirm-cancel',
+    animationDuration: 300,
+    showConfirmCallback: null,
+    messageSlotSelector: '#confirm-title',
+    contentSlots: {
+      body: {
+        contentAttribute: 'confirm-details',
+        slotSelector: '#confirm-body'
+      },
+      acceptText: {
+        contentAttribute: 'confirm-button',
+        slotSelector: '#confirm-accept'
+      }
+    }
+  }
+
+  constructor(options = {}) {
+    for(const [key, value] of Object.entries(options)) {
+      this.#config[key] = value
+    }
+
+    this.#controller = new ConfirmationController(this.#config.dialogSelector, this)
+  }
+
+  confirm(message, _formElement, submitter) {
+    const clickTarget = this.#clickTarget(submitter)
+    const contentMap = this.#contentMap(message, clickTarget)
+
+    return this.confirmWithContent(contentMap)
+  }
+
+  confirmWithContent(contentMap) {
+    return new Promise(resolve => this.#controller.showConfirm(contentMap, resolve))
+  }
+
+  showConfirm(element) {
+    element.classList.add(this.#config.activeClass)
+    // support dialog element
+    if (typeof element.showModal === 'function') {
+      element.showModal()
+    }
```

Comment:
considering moving this to be default value of `showConfirmCallback` config option..

Reply:
this along with doing the same for a new `hideConfirmCallback` would mean that we support the dialog element out of the box.  I've already added system tests ensuring a `<dialog>` based confirm works as expected.